save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/ContestController.kt

```diff
@@ -54,4 +70,24 @@ internal class ContestController(
     ): Flux<ContestDto> = Flux.fromIterable(
         contestService.findFinishedContests(pageSize)
     ).map { it.toDto() }
+
+    /**
+     * @param contestName
+     * @return Organization
+     */
+    @GetMapping("/{contestName}/public-test")
+    @PreAuthorize("permitAll()")
+    @Suppress("UnsafeCallOnNullableType")
+    fun getPublicTestForContest(
+        @PathVariable contestName: String,
+    ): Mono<PublicTestDto> {
+        val contest = contestService.findByName(contestName).get()
+        val testSuite = testSuitesService.findTestSuiteById(
+            contest.testSuiteIds?.split(",")?.first()?.toLong()!!
```

Comment:
This can be a `ContestService#getPublicTestSuiteIds` method for better readability. Also, why three safe-access operators followed by one `!!`, if the only place with nullability is the first `?.`?